src/main/app/controller/lists/ContactTypesController.ts

```diff
@@ -0,0 +1,163 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {ContactType} from '../../../types/ContactType';
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+
+@autobind
+export class ContactTypesController {
+
+  editContactTypeUrl = 'lists/contact-type/';
+  deleteContactTypeUrl = 'lists/contact-type/';
+  deleteConfirmUrl = 'lists/contact-type/delete-confirm/';
+
+  getContactTypesError = 'An error occurred when retrieving the contact types.';
+  getContactTypeError = 'A problem occurred when retrieving the contact type.';
+  updateContactTypeError = 'A problem occurred when trying to save the contact type.';
+  contactTypeAlreadyExistsError = 'A contact type with the proposed name already exists. The name must be unique.';
+  deleteError = 'A problem occurred when trying to delete the contact type.';
+  contactTypeInUseError = 'You cannot delete this contact type at the moment, as one or more courts are dependent on it. ' +
+    'Please remove the contact type from the relevant courts first';
+  nameRequiredError = 'Name is required.';
+
+  public async getAll(req: AuthedRequest, res: Response): Promise<void> {
+    await this.renderAll(req, res);
+  }
+
+  public async getContactType(req: AuthedRequest, res: Response): Promise<void> {
+    const errors: { text: string }[] = [];
+    const id = req.params?.id;
+    let fatalError = false;
+
+    let contactType: ContactType = null;
+    if (id) {
+      await req.scope.cradle.api.getContactType(id)
+        .then((value: ContactType) => contactType = value )
+        .catch(() => {
+          errors.push({ text: this.getContactTypeError });
+          fatalError = true;
+        });
+    }
+    this.renderContactType(res, contactType, false, errors, fatalError);
+  }
+
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+    let contactType = req.body.contactType as ContactType;
+
+    if(!CSRF.verify(req.body._csrf)) {
+      this.renderContactType(res, contactType, false, [{ text: this.updateContactTypeError }]);
+      return;
```

Comment:
Looks different to other examples, postcodes one for example is: 

    if (!CSRF.verify(req.body.csrfToken)) {
      return this.get(req, res, '', existingPostcodes, this.addErrorMsg, areasOfLaw, courtTypes);
    }

Should we have 

return this.renderContactType(res, contactType, false, [{ text: this.updateContactTypeError }]);

instead?

Reply:
does the same thing